### PR TITLE
Beta [v2.1]

### PR DESCRIPTION
**Pull request by itsjoshthedeveloper**
READY TO MERGE INTO `master`!

### CODE REVIEW

File: frontend/src/utilities.tsx [hunk 1]
```diff
@@ -34,6 +35,54 @@ export const useComponentVisible = <T extends HTMLElement>(
...
+export const useComponentVisibleDropdown = <T extends HTMLElement>(
+  initialIsVisible: boolean,
+  callback?: (visible: boolean) => void
+) => {
+  // Is the component visible?
+  const [isComponentVisible, setIsComponentVisible] = useState(
+    initialIsVisible
+  );
+  const ref_toggle = useRef<T>(null);
+  const ref_dropdown = useRef<T>(null);
```
Comment (ErikBoesen) [inline frontend/src/utilities.tsx:48]: Why snake_case?

File: frontend/src/Globals.jsx [hunk 2]
```diff
@@ -123,27 +124,29 @@ function Globals({ children }) {
...
-                </ThemeProvider>
-              </Router>
+              <SearchProvider>
+                <Router history={history}>
+                  <SPAPageChangeListener />
+                  <ThemeProvider theme={themeMode}>
+                    <>
+                      <GlobalStyles />
+                      <div id="base" style={{ height: 'auto' }}>
+                        {/* Clone child and give it the themeToggler prop */}
```
[the remark this answers is not in the document]
Comment (itsjoshthedeveloper) [reply frontend/src/Globals.jsx:134]: Ya you have to put comments in curly brackets in JSX

File: frontend/src/components/Popout.tsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+      return {
+        color: theme.primary_hover,
+      };
+    }
+    return undefined;
+  };
+
+  // Dynamically set popout button text based on selected options
+  useEffect(() => {
+    if (select_options) {
```
[the remark this answers is not in the document]
Comment (itsjoshthedeveloper) [reply frontend/src/components/Popout.tsx:146]: Yeah we probably should. We use a mix of camel and snake case throughout the codebase.. Maybe someone can go through it all later and clean it up?

File: frontend/src/utilities.tsx (hunk 1 above)
Comment (itsjoshthedeveloper) [reply frontend/src/utilities.tsx:48]: https://github.com/coursetable/coursetable/pull/611#discussion_r602939179

File: frontend/src/components/SearchResultsGridItem.js [hunk 4]
```diff
@@ -22,21 +22,25 @@ import { ReactComponent as Teacher } from '../images/catalog_icons/teacher.svg';
...
 const StyledGridItem = styled.div`
-  background-color: ${({ theme }) =>
-    theme.theme === 'light' ? 'rgb(245, 245, 245)' : theme.surface[1]};
-  transition: background-color 0.2s linear;
+  background-color: ${({ theme, inWorksheet }) =>
+    inWorksheet
+      ? theme.primary_light
+      : theme.theme === 'light'
+      ? 'rgb(245, 245, 245)'
+      : theme.surface[1]};
```
[the remark this answers is not in the document]
Comment (itsjoshthedeveloper) [reply frontend/src/components/SearchResultsGridItem.js:32]: prettier removes parentheses here automatically..

File: frontend/src/components/Themes.ts [hunk 5]
```diff
@@ -26,7 +38,14 @@ export const darkTheme: DefaultTheme = {
   select_hover: 'rgba(68, 100, 145, 0.75)', // Blueish tint to use on hover
   multivalue: '#4d4d4d', // Multivalue background-color used in react-select
   hidden: '#4d4d4d', // Color to use when course is hidden
+  disabled: 'rgba(242, 242, 242, 0.1)', // Disabled color for react-select
+  button_hover: 'rgba(255, 255, 255, 0.02)', // Button hover color
+  button_active: 'rgba(255, 255, 255, 0.05)', // Button active color (when pressed)
+  icon: 'rgb(102, 102, 102)', // Icon color
+  icon_focus: 'rgb(204, 204, 204)', // Icon focus color
```
[the remark this answers is not in the document]
Comment (itsjoshthedeveloper) [reply frontend/src/components/Themes.ts:45]: i wanted to more easily see the differences between related colors

**Comment by itsjoshthedeveloper** [comment]
@kevinhu the preview deploy is still redirecting to coursetable.com after cas authentication

**Comment by kevinhu** [comment]
The redirect is working on my end – try refreshing your DNS cache (or open in incognito for now)?

**Comment by itsjoshthedeveloper** [comment]
> The redirect is working on my end – try refreshing your DNS cache (or open in incognito for now)?

i tried both and it's still going to coursetable.com... yours is going to https://coursetable-fljczndgi-coursetable.vercel.app/catalog?

**Comment by kevinhu** [comment]
Nah, mine is just going back to beta.coursetable.com – which browser are you using?

**Comment by itsjoshthedeveloper** [comment]
Chrome. isn't it supposed to go to https://coursetable-fljczndgi-coursetable.vercel.app/catalog for the preview deploy (7d7a1ee0)?

**Comment by kevinhu** [comment]
Ah, so I haven't configured it yet to work with all deploys on the Vercel domain – however, I have set up this beta branch specifically to deploy on beta.coursetable.com. I'm working on the additional deploy previews now.

**Comment by kevinhu** [comment]
Update: API has been successfully updated and authentication now works on all preview deployments.

**Pull request merged**